server/src/frameworks/Hardhat/HardhatProject.ts

```diff
@@ -253,6 +243,32 @@ export class HardhatProject extends Project {
     return resolveActionsFor(this.serverState, diagnostic, document, uri);
   }
 
+  private _sendToChild(message: Message) {
+    this._assertWorkerExists();
+    this._assertWorkerIsRunning();
```

Comment:
We only want to be sending if the child process is considered running; if it is initializing, errored or stopped we should not be sending a message.

Reply:
I removed the `_assertWorkerIsRunning` check. Looking through the code I can't predict if there are places where we don't make this assumption.